game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProMoveUtils.java

```diff
@@ -95,52 +97,48 @@ public static void calculateMoveRoutes(
 
           // Sea unit (including carriers with planes)
           route =
-              data.getMap()
-                  .getRouteForUnit(
-                      startTerritory,
-                      t,
-                      ProMatches.territoryCanMoveSeaUnitsThrough(player, data, isCombatMove),
-                      u,
-                      player);
+              map.getRouteForUnit(
+                  startTerritory,
+                  t,
+                  ProMatches.territoryCanMoveSeaUnitsThrough(player, data, isCombatMove),
+                  u,
+                  player);
         } else if (!unitList.isEmpty() && unitList.stream().allMatch(Matches.unitIsLand())) {
 
           // Land unit
           route =
-              data.getMap()
-                  .getRouteForUnit(
-                      startTerritory,
-                      t,
-                      ProMatches.territoryCanMoveLandUnitsThrough(
-                          player, data, u, startTerritory, isCombatMove, new ArrayList<>()),
-                      u,
-                      player);
+              map.getRouteForUnit(
+                  startTerritory,
+                  t,
+                  ProMatches.territoryCanMoveLandUnitsThrough(
+                      player, data, u, startTerritory, isCombatMove, List.of()),
```

Comment:
Did you double check that all these empty lists that are passed in are never modified since I believe `List.of()` is an immutable list and `new ArrayList<>()` isn't?

Reply:
It's just used for territoryCanMoveLandUnitsThrough() as the enemyTerritories param. And that's just used for:
          .and(Matches.territoryIsInList(enemyTerritories).negate())

So there's no problem.